File: Core/Components/GridViewItem.cs
using Bridge.Html5;
using Core.Extensions;
using Core.Models;
using Core.MVVM;
using System.Collections.Generic;
using System.Linq;
using ElementType = Core.MVVM.ElementType;

namespace Core.Components
{
    public class GridViewItem : ListViewItem
    {
        public GridViewItem(ElementType tr) : base(tr)
        {
        }

        internal override void RenderRowData(List<Component> headers, object row, int? index = null, bool emptyRow = false)
        {
            if (index.HasValue)
            {
                if (index >= Element.ParentElement.Children.Count() || index < 0)
                {
                    index = 0;
                }

                Element.ParentElement.InsertBefore(Element, Element.ParentElement.Children[index.Value]);
            }
            headers.Where(x => !x.Hidden).ForEach(header =>
            {
                RenderTableCell(row, header, null);
            });
            BindingEvents();
        }

        internal override void RenderTableCell(object rowData, Component header, HTMLElement cellWrapper = null)
        {
            Html.Take(Element).TData.TabIndex(-1)
                .Event(EventType.FocusIn, (e) => FocusCell(e, header))
                .DataAttr("field", header.FieldName).Render();
            if (header.StatusBar)
            {
                Html.Instance.Icon("fa fa-pencil").End.Render();
            }
            if (string.IsNullOrEmpty(header.FieldName))
            {
                return;
            }
            base.RenderTableCell(rowData, header, Html.Context);
            Html.Instance.EndOf(ElementType.td);
        }

        private void FocusCell(Event e, Component header)
        {
            ListViewSection.ListView.LastElementFocus?.Closest(ElementType.td.ToString()).RemoveClass("cell-selected");
            var td = e.Target as HTMLElement;
            td.Closest(ElementType.td.ToString()).AddClass("cell-selected");
            ListViewSection.ListView.LastElementFocus = td;
            ListViewSection.ListView.LastComponentFocus = header;
            ListViewSection.ListView.EntityFocusId = Entity[IdField]?.ToString();
        }
    }
}

File: CoreAPI/wwwRoot/js/core/gridViewItem.js
import { Component } from "./models/component.js";
import { ListViewItem } from "./listViewItem.js";
import { ElementType } from "./models/elementType.js";
import EventType from "./models/eventType.js";
import { Html } from "./utils/html.js";

export class GridViewItem extends ListViewItem {
    constructor(tr) {
        super(tr);
    }

    /**
     * 
     * @param {Component[]} headers 
     * @param {any} row 
     * @param {number} index 
     * @param {*} emptyRow 
     */
    RenderRowData(headers, row, index = null, emptyRow = false) {
        if (index !== null) {
            if (index >= this.Element.parentElement.children.length || index < 0) {
                index = 0;
            }

            this.Element.parentElement.insertBefore(this.Element, this.Element.parentElement.children[index]);
        }

        headers.filter(x => !x.Hidden).forEach(header => {
            this.RenderTableCell(row, header, null);
        });
        this.BindingEvents();
    }

    /**
     * @param {any} rowData
     * @param {Component} header
     */
    RenderTableCell(rowData, header, cellWrapper = null) {
        Html.Take(this.Element).TData.TabIndex(-1)
        .Event(EventType.FocusIn, (e) => this.FocusCell(e, header))
        .DataAttr("field", header.FieldName).Render();
    
        if (header.StatusBar) {
            Html.Instance.Icon('fa fa-pencil').End.Render();
        }
    
        if (!header.FieldName) {
            return;
        }
    }
    
    /**
     * @param {Event} e
     * @param {Component} header
     */
    FocusCell(e, header) {
        if (this.ListViewSection == null) return;
        if (this.ListViewSection.ListView.LastElementFocus) {
            this.ListViewSection.ListView.LastElementFocus?.Closest(ElementType.td.toString()).RemoveClass("cell-selected");
        }
    
        /** @type {HTMLTableCellElement} */
        // @ts-ignore
        let td = e.target;
        td.closest(ElementType.td).classList.add("cell-selected");
        this.ListViewSection.ListView.LastElementFocus = td;
        this.ListViewSection.ListView.LastComponentFocus = header;
        this.ListViewSection.ListView.EntityFocusId = this.EntityId;
    }
}